curvefs/src/mds/topology/topology.cpp

```diff
@@ -1107,11 +1144,28 @@ TopoStatusCode TopologyImpl::ChooseSingleMetaServerInZone(
                    << "zoneId = " << zoneId;
         return TopoStatusCode::TOPO_METASERVER_NOT_FOUND;
     }
-    auto iter = metaServerList.begin();
-    int randomValue = GetOneRandomNumber(0, metaServerList.size() - 1);
-    std::advance(iter, randomValue);
-    *metaServerId = *iter;
-    return TopoStatusCode::TOPO_OK;
+
+    while (metaServerList.size() > 0) {
+        auto iter = metaServerList.begin();
+        int randomValue = GetOneRandomNumber(0, metaServerList.size() - 1);
+        std::advance(iter, randomValue);
+        // check metaserver online status
+        MetaServer metaserver;
+        if (GetMetaServer(*iter, &metaserver)) {
+            // TODO(wanghai01): consider the space used by per copyset more details later  // NOLINT
```

Comment:
metaserver mainly use memory instead of disk

Reply:
> metaserver mainly use memory instead of disk

Yes, but now there is no memory capacity in metaserver heartbeat.